py_gtfs_rt_ingestion/lib/postgres_utils.py

```diff
@@ -187,28 +182,24 @@ def _rds_writer_process(metadata_queue: Queue) -> None:
             "metadata_insert", filepath=metadata_path
         )
         process_logger.log_start()
-        for retry_attempt in range(retry_count):
+        retry_attempt = 0
+
+        # All metatdata_insert attempts must succeed, keep attempting until success
+        while True:
             try:
-                process_logger.add_metadata(retry_attempts=retry_attempt)
                 with engine.begin() as cursor:
                     cursor.execute(insert_statement)
 
-            except Exception as exception:
-                if retry_attempt == retry_count - 1:
-                    bad_insert += 1
-                    process_logger.log_failure(exception)
-                else:
-                    # wait for gremlins to disappear
-                    time.sleep(15)
+            except Exception as _:
+                retry_attempt += 1
+                # wait for gremlins to disappear
+                time.sleep(15)
```

Comment:
Change this to unlimited retry attempts. This should never fail unless RDS is having issues, in which case we should keep the queue of metadata files to write and write them when RDS connection is restored.